### PR TITLE
chore!: Normalize repository, dropping node <10.13 support

### LINKED ISSUE #15: Normalize repository, dropping node <10.13
@sttk want to take this on?

### PR DESCRIPTION
**Pull request by sttk**
This PR updates this project:

- Updates dependency versions
- Introduces GitHub Actions
- Adds, changes and deletes some project files

I separated testing to `npm test` and `npm run cover` because one test case does not work on nyc.

Closes #15

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) 2014-2018 Tyler Kellen <tyler@sleekcode.net>, Blaine Bublitz <blaine.bublitz@gmail.com>, and Eric Schoffstall <yo@contra.io>
+Copyright (c) 2014-2018, 2021 Tyler Kellen <tyler@sleekcode.net>, Blaine Bublitz <blaine.bublitz@gmail.com>, Eric Schoffstall <yo@contra.io> and other contributors
```
Comment (phated) [inline LICENSE]:
```suggestion
Copyright (c) 2014-2018, 2021 Tyler Kellen <tyler@sleekcode.net>, Blaine Bublitz <blaine.bublitz@gmail.com>, and Eric Schoffstall <yo@contra.io>
```
Let's keep this to just the 3.

File: package.json [hunk 2]
```diff
@@ -24,21 +24,27 @@
     "lint": "eslint .",
     "pretest": "npm run lint",
     "test": "mocha --async-only",
-    "cover": "nyc --reporter=lcov --reporter=text-summary npm test",
-    "coveralls": "npm run cover && istanbul-coveralls",
+    "cover": "nyc mocha --async-only",
```
Comment (phated) [inline package.json]: I think this is wrong. We move the nyc into the "test" script, right?

File: package.json [hunk 3]
```diff
@@ -24,21 +24,27 @@
...
-    "mocha": "^3.5.3",
-    "nyc": "^11.3.0",
-    "v8flags": "^3.0.1"
+    "eslint": "^7.32.0",
+    "eslint-config-gulp": "^5.0.1",
+    "eslint-plugin-node": "^11.1.0",
+    "expect": "^27.3.1",
+    "mocha": "^8.4.0",
+    "nyc": "^15.1.0",
+    "v8flags": "^3.2.0"
```
Comment (phated) [inline package.json]: We'll need to release this and update to the next major

File: test/index.js [hunk 4]
```diff
@@ -121,7 +121,7 @@ describe('flaggedRespawn', function() {
 
     it('should respawn; if child is killed, parent should exit with same signal', function(done) {
       // Because travis and nyc hates this
-      if (process.env.TRAVIS || process.env.NYC_PARENT_PID) {
+      if (process.env.TRAVIS || process.env.NYC_PARENT_PID || process.env.NYC_PROCESS_ID) {
```
Comment (phated) [inline test/index.js]: We can remove the TRAVIS one, I think
```suggestion
      if (process.env.NYC_PARENT_PID || process.env.NYC_PROCESS_ID) {
```

File: package.json [hunk 5]
```diff
@@ -24,21 +24,27 @@
     "lint": "eslint .",
     "pretest": "npm run lint",
     "test": "mocha --async-only",
-    "cover": "nyc --reporter=lcov --reporter=text-summary npm test",
-    "coveralls": "npm run cover && istanbul-coveralls",
+    "cover": "nyc mocha --async-only",
     "respawn": "node test/bin/respawner --harmony test",
     "nospawn": "node test/bin/respawner test"
```
Comment (phated) [inline package.json]: How are these used?

**Review by phated — COMMENTED** [review]
Thanks for looking into this @sttk! I had a few things.

File: package.json (hunk 3 above)
Comment (phated) [reply package.json]: I just released v8flags v4.0.0

File: LICENSE (hunk 1 above)
Comment (sttk) [reply LICENSE:3]: Got it. I'll modify it.

File: package.json (hunk 2 above)
Comment (sttk) [reply package.json]: Got it. I'll modify it.

File: package.json (hunk 3 above)
Comment (sttk) [reply package.json]: I got it.

File: package.json (hunk 5 above)
Comment (sttk) [reply package.json]: I think they aren't used, too. I'll remove them.

**Comment by sttk** [comment]
@phated After using `v8flag` v4 the tests failed, because `flagged-respawn` processes on the premise that the separator of v8 options in the cache file is underscore, but became dash by [this change](https://github.com/gulpjs/v8flags/commit/e5194ca0b4e5e3d4f28b5a42903a20b477ee3eb9#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346L79).

For compatibility of the cache file, shouldn't we keep this separator underscore?

And Node.js>=v10 also accepts both dash and underscore as v8 option's separator.

**Comment by sttk** [comment]
Sorry, we should use dash and underscore separately depending on `CACHE_VERSION` of `v8flag` cache file.
I'll change about this.

**Comment by phated** [comment]
@sttk this will be a semver major version increase, so we can stop supporting the snake_case flags. Even though node still supports these flags, we only supported them because old versions of node did not support hyphens.

**Comment by sttk** [comment]
@phated I got it. I'll remove the codes which replace dashes in v8 flags to underscores.

**Comment by sttk** [comment]
@phated I've modified what you pointed out. Please review them.

**Review by phated — APPROVED** [review]
Excellent work! Thanks @sttk 🎊

**Comment by sttk** [comment]
@phated Thank you for merging!

**Pull request merged**